processing/src/main/java/io/druid/segment/IndexSpec.java

```diff
@@ -91,7 +94,8 @@ public IndexSpec()
   public IndexSpec(
       @JsonProperty("bitmap") BitmapSerdeFactory bitmapSerdeFactory,
       @JsonProperty("dimensionCompression") String dimensionCompression,
-      @JsonProperty("metricCompression") String metricCompression
+      @JsonProperty("metricCompression") String metricCompression,
+      @JsonProperty("metricLongEncoding") String metricLongEncoding
```

Comment:
@xvrl I think it's a good idea to have a separate set of encodings for each data type since encoding is generally not applicable to multiple data types. If this is the case, it doesn't make much sense to have a common metricEncoding value since no type specific encoding can be used.

For the acceptable encodings, I followed the same way as CompressionStrategy, where acceptable values are lowercase string of all LongEncodingFormat enum values.